### ISSUE TITLE
Batch 1: RNN/Sequence + Optimizer specs

### ISSUE THREAD

**Issue by gHashTag** [gHashTag/trinity#489]
## Task

Create 8 RNN/Sequence and Optimizer .tri specifications for CLARA TA1.

**Why These 8 Specs**
- Batch 2: Adam, RMSprop (need  complement Batch 1)
- Optimizers are critical for training loops

**Deadline**: April 9, 2026 (DARPA TA1 requirement)

## Specs to Create

| File | Description | Priority |
|------|-------------|----------|
| `specs/algo/adam.tri` | Adam optimizer with momentum | P0 |
| `specs/algo/rmsprop.tri` | RMSprop optimizer | P0 |
| `specs/algo/mse_loss.tri` | MSE loss function | P0 |
| `specs/algo/huber_loss.tri` | Huber loss (robust regression) | P0 |
| `specs/algo/kl_divergence.tri` | KL divergence (VAE/GANs) | P0 |
| `specs/algo/contrastive_loss.tri` | Contrastive loss (self-supervised) | P0 |

## Templates (reuse from existing specs)

All specs will follow this template structure:

``yaml
name: {name}
version: "1.0.0"
module: algo.optimizer.{name}
description: "{description}"

types:
  {Type}Config:
    description: "{name} configuration"
    fields:
      - name: learning_rate
        type: f32
        description: "Learning rate for gradient updates"

functions:
  {fn_name}:
    params: [...params]
    returns: {return_type}
    description: "{formula}"
``

## Dependencies

- Blocked by: #486 (CLARA demo needs these specs + Batch 1)
- Related: Batch 1 RNN specs

## Acceptance

- [ ] 8 .tri files created
- [ ] `tri gen` produces stubs (pending full VIBEE)
- [ ] Sacred constants verified
- [ ] Xavier initialization patterns

## Next Steps

1. Create 8 spec files
2. Test `tri gen` compilation
3. Commit to issue #487
